### PR TITLE
Add flutter attach documentation

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

The issue below highlights the fact that our `flutter attach` command is not well documented. This PR adds a markdown document which describes the functionality of the command. I intend to link to this markdown document from the wiki once the PR lands in master.

## Related Issues

[`flutter help attach` should provide better instructions #30826](https://github.com/flutter/flutter/issues/30826)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter_tools/doc/attach.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+to lookup the observatory port via the application ID, with just `$ flutter
+attach`
+1. If the platform is Fuchsia and the app is not yet running, the module name
+must be provided, e.g. `$ flutter attach --module=mod_name`. Attach will then
+listen and wait for the mod to start
+1. On other platforms (i.e. Android), if the app is not yet running attach
+will listen and wait for the app to be (manually) started with the default
+command: `$ flutter attach`
+
+## To-do
```
Comment (zanderso) [inline packages/flutter_tools/doc/attach.md]: If we don't already have an issue filed for this, then that's where this should go instead of here.

**Review by zanderso — COMMENTED** [review]
Nice!

File: packages/flutter_tools/doc/attach.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+There are four ways for the attach command to discover a running app:
+
+1. If the app is already running and the observatory port is known, it can be
+explicitly provided to attach via the command-line, e.g. `$ flutter attach
+--debug-port 12345`
+1. If the app is already running and the platform is iOS, attach can use mDNS
+to lookup the observatory port via the application ID, with just `$ flutter
+attach`
+1. If the platform is Fuchsia and the app is not yet running, the module name
```
Comment (jonahwilliams) [inline packages/flutter_tools/doc/attach.md]: This can also be invoked after the app is running

File: packages/flutter_tools/doc/attach.md (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/doc/attach.md]: We have [this issue](https://github.com/flutter/flutter/issues/23687), so I will remove this text.

File: packages/flutter_tools/doc/attach.md (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/doc/attach.md]: updated

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**